python/lsst/ts/externalscripts/auxtel/latiss_acquire.py

```diff
@@ -0,0 +1,604 @@
+        for iter_num in range(self.max_acq_iter):
+            self.log.debug(
+                f"\nStarting iteration number {iter_num + 1}, with a "
+                f"maximum of {self.max_acq_iter}"
+            )
+
+            self.latiss.rem.atoods.evt_imageInOODS.flush()
+            tmp = await self.latiss.take_acq(
+                exptime=self.acq_exposure_time,
+                n=1,
+                group_id=self.group_id,
+                reason=self.reason,
+                program=self.program,
+            )
+            data_id = await self.get_next_image_data_id(
+                timeout=self.acq_exposure_time + STD_TIMEOUT, flush=False
+            )
+            self.log.debug(
+                f"Take Object returned {tmp}. Now waiting for image to land in OODS"
+            )
+
+            dx_arcsec, dy_arcsec = await self.find_offset(data_id, target_position)
+
+            dr_arcsec = np.sqrt(dx_arcsec**2 + dy_arcsec**2)
+
+            self.log.debug(
+                f"Calculated offsets [dx,dy] are [{dx_arcsec:0.2f}, {dy_arcsec:0.2f}] arcsec as calculated"
+                f" from sequence number {data_id['seq_num']} on Observation Day of {data_id['day_obs']}"
+            )
+
+            # Check if star is in place, if so then we're done
+            if dr_arcsec < self.target_pointing_tolerance:
+                self.log.info(
+                    "Acquisition completed successfully."
+                    f"Current radial pointing error of {dr_arcsec:0.2f} arcsec is within the tolerance "
+                    f"of {self.target_pointing_tolerance} arcsec. "
+                )
+                _success = True
+                break
+            else:
+                self.log.info(
+                    f"Current radial pointing error of {dr_arcsec:0.2f} arcsec exceeds the tolerance"
+                    f" of {self.target_pointing_tolerance} arcsec."
+                )
+
+                # Offset telescope, using only relative offsets
+                self.log.info("Applying x/y offset to telescope pointing.")
+
+                # Use persistent = False otherwise when we switch gratings
+                # it may keep an offset we no longer want
+                await self.atcs.offset_xy(
+                    x=dx_arcsec, y=dy_arcsec, relative=True, persistent=False
+                )
+
+            self.log.debug(
+                f"At end of iteration loop {iter_num+1}, success is {_success}."
+            )
+
+        # Check that maximum number of iterations for acquisition
+        # was not reached
+        if not _success:
+            raise RuntimeError(
+                f"Failed to acquire star on target after {iter_num} images."
+            )
+
+    async def latiss_slew_and_acquire(self):
+        """Performs slew to new target specified in configuration and initiates
+        performs acquisition loop. A final, verification image is taken if
+        configured.
+
+        Parameters
```

Comment:
Remove Parameters and Returns sessions if there are no input parameters or return values.